src/SprykerEco/Zed/Econda/Business/EcondaBusinessFactory.php

```diff
@@ -0,0 +1,285 @@
+    /**
+     * @return \Generated\Shared\Transfer\BatchResultTransfer
+     */
+    protected function createBatchResultModel()
+    {
+        return new BatchResultTransfer();
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Econda\Business\Helper\ProgressBarHelper
+     */
+    protected function createProgressBarHelper()
+    {
+        return new ProgressBarHelper();
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Econda\Business\Exporter\Writer\File\NameGenerator\CsvNameGenerator
+     */
+    protected function createCsvNameGenerator()
+    {
+        return new CsvNameGenerator();
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Econda\Business\Collector\File\EcondaCategoryCollector
+     */
+    public function createFileCategoryCollector()
+    {
+        $storageCategoryNodeCollector = new EcondaCategoryCollector(
+            $this->createCriteriaBuilder(),
+            $this->createStoragePdoQueryAdapterByName('CategoryNodeEcondaQuery')
+        );
+
+        return $storageCategoryNodeCollector;
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Econda\Business\Collector\File\EcondaProductCollector
+     */
+    public function createFileProductCollector()
+    {
+        /**
+         * @var \SprykerEco\Zed\Econda\Business\Collector\File\EcondaProductCollector $storageProductCollector
+         */
+        $storageProductCollector = new EcondaProductCollector(
+            $this->createCriteriaBuilder(),
+            $this->createStoragePdoQueryAdapterByName('ProductConcreteEcondaQuery'),
+            $this->getProductCategoryQueryContainer(),
+            $this->getProductImageQueryContainer(),
+            $this->getPriceFacade(),
+            $this->getConfig()
+        );
+
+        return $storageProductCollector;
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Econda\Business\Exporter\ExporterInterface
+     */
+    protected function createFileExporter()
+    {
+        $fileExporter = new FileExporter(
+            $this->createFileWriter(),
+            $this->createFailedResultModel(),
+            $this->createBatchResultModel(),
+            $this->createCsvNameGenerator(),
+            $this->getConfig()->getFileExportPath(),
+            $this->getCollectorFileExporterPlugins()
+        );
+
+        return $fileExporter;
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Econda\Dependency\Plugin\EcondaPluginInterface[]
+     */
+    protected function getCollectorFileExporterPlugins()
+    {
+        return $this->getProvidedDependency(EcondaDependencyProvider::FILE_PLUGINS);
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Econda\Dependency\Facade\EcondaToLocaleInterface
+     */
+    protected function getLocaleFacade()
+    {
+        return $this->getProvidedDependency(EcondaDependencyProvider::FACADE_LOCALE);
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Econda\Persistence\EcondaQueryContainer
+     */
+    protected function getEcondaQueryContainer()
+    {
+        return new EcondaQueryContainer();
```

Comment:
This would not be get but create.
BUT the query container does not have to be "created", it can and should be passed from the DP if external, otherwise it already is available here internal (see doc block annotations).